ice/http/response.zep

```diff
@@ -175,6 +184,178 @@ class Response implements ResponseInterface
+     * ----------|-----------|------------------------------------|--------------
+     * string    | file      | file that already exists           | null
+     * boolean   | inline    | Display inline instead of download | FALSE
+     * boolean   | resumable | Allow to resumable download        | FALSE
+     * boolean   | delete    | Delete the file after sending      | FALSE
+     * int       | timeout   | Execute time for the script        | 0
+     * int       | speed     | Download speed in millisecond      | 0
+     *
+     * Download a file that already exists:
+     *
+     *     $request->sendFile('ice.zip', 'application/zip', ['file' => '/download/latest.zip']);
+     *
+     * Download generated content as a file:
+     *
+     *     $response->setContent($content);
+     *     $response->sendFile($filename, $mineType);
+     *
+     * Attention: No further processing can be done after this method is called!
+     *
+     * @param string filename The file name of the attachment
+     * @param string mime Manual mime type
+     * @param array options The keys can be [file|inline|resumable|delete|timeout|speed]
+     * @return  void
+     */
+    public function sendFile(string filename, string mime, array options = [])
+    {
+        var file, filepath, data, size, isDelete, disposition, block, pos, speed;
+
+        if empty options["file"] {
+            // Force the data to be rendered if
+            let data = (string) this->body;
+
+            // Get the content size
+            let size = strlen(data);
+
+            // Create a temporary file to hold the current response
+            let file = tmpfile();
+
+            // Write the current response into the file
+            fwrite(file, data);
+
+            // File data is no longer needed
+            unset(data);
+
+            let isDelete = true;
+        } else {
+            // Get the complete file path
+            let filepath = realpath(options["file"]);
+
+            // Get the file size
+            let size = filesize(filepath);
+
+            // Open the file for reading
+            let file = fopen(filepath, 'rb');
+
+            fetch isDelete, options["delete"];
+        }
+
+        if !is_resource(file) {
+            throw new Exception(['Could not read file to send: %s',  filename]);
+        }
+
+        if empty options["inline"] {
+            let disposition = "attachment";
+        } else {
+            let disposition = "inline";
+        }
+
+        if empty options["resumable"] {
+            this->headers->set("Content-Length", size);
+        } else {
+            // Calculate byte range to download.
+            let range = this->getByteRange(size),
+                start = range[0],
+                end = range[1];
+
+            // HTTP/1.1 416 Requested Range Not Satisfiable
+            if range[2] != null {
+                header(range[2]);
+                exit();
```

Comment:
Could we return the response _416 Requested Range Not Satisfiable_ here?
```zep
this->setStatus(416);
this->setBody(response->getMessage(416));
return this;
```

Reply:
bad request MUST stop developer to response any other content to user